sys/include/net/gnrc/csma_sender.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2015 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_csma_sender  Helper interface to send packets via CSMA/CA
+ * @ingroup     net_gnrc
+ * @brief       This interface allows code from layer 2 (MAC) or higher
+ *              to send packets with CSMA/CA, whatever the abilities and/or
+ *              configuration of a given radio transceiver device are.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the CSMA/CA helper
+ *
+ * @author      Kévin Roussel <Kevin.Roussel@inria.fr>
+ */
+
+#ifndef GNRC_CSMA_SENDER_H_
+#define GNRC_CSMA_SENDER_H_
+
+#include "kernel.h"
+#include "net/gnrc.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief Default Minimal CSMA/CA Backoff Exponent 
+ */
+#define MAC_MIN_BE_DEFAULT               (3U)
+
+/**
+ * @brief Default Maximal CSMA/CA Backoff Exponent 
+ */
+#define MAC_MAX_BE_DEFAULT               (5U)
+
+/**
+ * @brief Default Maximal number of retries for sending
+ *        a given packet with the CSMA/CA method
+ */
+#define MAC_MAX_CSMA_BACKOFFS_DEFAULT    (4U)
+
+/**
+ * @brief CSMA/CA backoff period, in microseconds
+ */
+#define A_UNIT_BACKOFF_PERIOD_MICROSEC   (320U)
+
+/**
+ * @brief Define a different (non-standard) value for
+ *        the CSMA macMinBE parameter.
+ *
+ * @param[in] val     new value for macMinBE
+ */
+void set_csma_mac_min_be(uint8_t val);
+
+/**
+ * @brief Define a different (non-standard) value for
+ *        the CSMA macMaxBE parameter.
+ *
+ * @param[in] val     new value for macMaxBE
+ */
+void set_csma_mac_max_be(uint8_t val);
+
+/**
+ * @brief Define a different (non-standard) value for
+ *        the CSMA macMaxCSMABackoffs parameter.
+ *
+ * @param[in] val     new value for macMaxCSMABackoffs
+ */
+void set_csma_mac_max_csma_backoffs(uint8_t val);
+
+/**
+ * @brief   Sends a 802.15.4 frame using the CSMA/CA method
+ *
+ * If the transceiver can (and is configured to) do hardware-assisted
+ * CSMA/CA, this feature is used. Otherwise, a software procedure is used.
+ *
+ * @param[in] dev       netdev device, needs to be already initialized
+ * @param[in] pkt       pointer to the data in the packet buffer;
+ *                      it must be a complete 802.15.4-compliant frame,
```

Comment:
Sorry to be picky here, but in fact the only requirement is that the payload is smaller than a certain threshold to fit into 802.15.4 frame, but apart from that it must only have a proper NETIF header which is agnostic of the underlying network. Especially you don't have to mess with 802.15.4 header creation yourself. Sorry for the late comment on that.